src/models/room.ts

```diff
@@ -1216,6 +1216,13 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
         return Array.from(this.threads.values());
     }
 
+    /**
+     * @experimental
+     */
+    public getThreadsMap(): Map<string, Thread> {
+        return this.threads;
+    }
```

Comment:
The alternative to this is using `getThreads()` but this just returns unhelpful `[[], []]` blank arrays of arrays if you haven't opened the threads yet.

It's useful to see which events are threads so we can return the raw map which has a map of `threadID` -> `Thread`

```json
{
    "$BWFwOdh4QmmUpEHNdjAYXEXM5jue3ZcTtLKhc7YIusw": [],
    "$VknYvjxNw2HKifHp_8C5F_WqDMXBTfdDYYT3-AVnNhY": ["$VknYvjxNw2HKifHp_8C5F_WqDMXBTfdDYYT3-AVnNhY","$c-6O4zVYEpEuXU5bWfqFtJXXKXARxJDn64BO_9HnP4A"]
}
```

Reply:
Your snippet shows an object string->string[]

But the map is threadId -> Thread

The key in the object id Thread::id so you gain 0 additional information from having the map vs the array of threads

Reply:
Ahh cool, `thread.id` is perfect 👌